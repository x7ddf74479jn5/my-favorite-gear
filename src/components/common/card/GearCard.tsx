import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardHeader from "@material-ui/core/CardHeader";
import IconButton from "@material-ui/core/IconButton";
import Link from "@material-ui/core/Link";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import ArrowDownwardIcon from "@material-ui/icons/ArrowDownward";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import DeleteIcon from "@material-ui/icons/Delete";
import LinkIcon from "@material-ui/icons/Link";
import ThumbUpIcon from "@material-ui/icons/ThumbUp";
import { brandColor } from "asset/variables";
import ReviewStars from "components/common/card/ReviewStars";
import type { FC } from "react";
import React from "react";
import type { Gear } from "services/models/gear";

interface GearCardProps {
  gear: Gear;
  addButton?: (gear: Gear) => void;
  favoriteList?: Gear[];
  removeButton?: (gear: Gear) => void;
  upButton?: (gear: Gear) => void;
  downButton?: (gear: Gear) => void;
}

const useStyles = makeStyles((theme) => {
  return {
    cardRoot: {
      width: "100%",
      margin: theme.spacing(1, 0),
    },
    avatar: {
      width: theme.spacing(9),
      height: theme.spacing(9),
    },
    content: {
      display: "flex",
      justifyContent: "space-between",
    },
    review: {
      display: "flex",
    },
    linkContainer: {
      display: "flex",
      gap: theme.spacing(2),
      margin: theme.spacing(2, 0, 2, 0),
    },
    amazon: {
      color: "white",
      backgroundColor: brandColor.amazon,
    },
    rakuten: {
      color: "white",
      backgroundColor: brandColor.rakuten,
    },
  };
});

const GearCard: FC<GearCardProps> = ({
  gear,
  addButton,
  favoriteList,
  removeButton,
  upButton,
  downButton,
}) => {
  const classes = useStyles();
  // const makeStarCount = () => {
  //   if (gear.reviewAverage === null) return;
  //   const integerPart = Math.floor(gear.reviewAverage);
  //   const decimalPart = gear.reviewAverage - integerPart;
  //   let starCount =
  //     decimalPart < 0.25
  //       ? integerPart
  //       : decimalPart < 0.5
  //       ? integerPart + decimalPart
  //       : decimalPart < 0.75
  //       ? integerPart + decimalPart
  //       : Math.ceil(gear.reviewAverage);
  //   const stars = [];
  //   while (starCount > 0) {
  //     if (starCount < 1) {
  //       stars.push(<StarHalfIcon key={starCount} />);
  //       break;
  //     }
  //     stars.push(<StarIcon key={starCount} />);
  //     starCount--;
  //   }

  //   return stars;
  // };
  // const stars = makeStarCount();

  return (
    <Card className={classes.cardRoot}>
      <CardHeader
        avatar={
          gear.mediumImageUrl ? (
            <Avatar
              src={gear.mediumImageUrl}
              alt={gear.productName}
              className={classes.avatar}
            />
          ) : (
            <Avatar className={classes.avatar}>{gear.productName}</Avatar>
          )
        }
        title={gear.productName}
        subheader={`${gear.makerName}/${gear.brandName}/${gear.genreName}`}
      />
      <CardContent>
        <div className={classes.content}>
          {gear.reviewAverage && (
            <div className={classes.review}>
              <ReviewStars reviewAverage={gear.reviewAverage} />
              <Typography> : {gear.reviewAverage}</Typography>
            </div>
          )}
          {gear.averagePrice && (
            <Typography align="right">
              平均価格: {gear.averagePrice}円
            </Typography>
          )}
        </div>
        <div className={classes.linkContainer}>
          {gear.affiliateUrl && (
            <Button
              className={classes.rakuten}
              component={Link}
              href={gear.affiliateUrl}
              startIcon={<LinkIcon />}
              disableElevation
              target="_blank"
              rel="noopener"
              fullWidth
            >
              楽天で見る
            </Button>
          )}
          {gear.affiliateUrl && (
            <Button
              className={classes.amazon}
              component={Link}
              href={gear.amazonUrl}
              startIcon={<LinkIcon />}
              disableElevation
              target="_blank"
              rel="noopener"
              fullWidth
            >
              Amazonで見る
            </Button>
          )}
        </div>
        {upButton && (
          <IconButton
            onClick={() => {
              return upButton(gear);
            }}
            size="small"
          >
            <ArrowUpwardIcon />
          </IconButton>
        )}
        {downButton && (
          <IconButton
            onClick={() => {
              return downButton(gear);
            }}
            size="small"
          >
            <ArrowDownwardIcon />
          </IconButton>
        )}
        {addButton ? (
          favoriteList &&
          favoriteList.filter((favoriteListGear) => {
            return favoriteListGear.productId === gear.productId;
          }).length > 0 ? (
            <Button
              color="secondary"
              startIcon={<ThumbUpIcon />}
              disableElevation
              disabled
            >
              My Favorite Gearに登録済み
            </Button>
          ) : (
            <Button
              color="secondary"
              startIcon={<ThumbUpIcon />}
              disableElevation
              onClick={() => {
                return addButton(gear);
              }}
            >
              My Favorite Gearに登録
            </Button>
          )
        ) : (
          <></>
        )}
        {removeButton ? (
          <Button
            color="inherit"
            startIcon={<DeleteIcon />}
            disableElevation
            onClick={() => {
              return removeButton(gear);
            }}
          >
            My Favorite Gearから削除
          </Button>
        ) : (
          <></>
        )}
      </CardContent>
    </Card>
  );
};

export default GearCard;
