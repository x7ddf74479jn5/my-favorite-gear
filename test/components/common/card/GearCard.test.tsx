import userEvent from "@testing-library/user-event";
import GearCard from "components/common/card/GearCard";
import React from "react";

import { render, testFavoriteList, testGear } from "../../../test-utils";

const mockAddButton = jest.fn();
const mockRemoveButton = jest.fn();
const mockUpButton = jest.fn();
const mockDownButton = jest.fn();

describe("GearCard", () => {
  it("matches snapshot", () => {
    let renderResult;

    renderResult = render(
      <GearCard
        gear={testGear}
        addButton={mockAddButton}
        removeButton={mockRemoveButton}
        upButton={mockUpButton}
        downButton={mockDownButton}
      />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();

    renderResult = render(
      <GearCard
        gear={testFavoriteList.gears[0]}
        favoriteList={testFavoriteList.gears}
        addButton={mockAddButton}
        removeButton={mockRemoveButton}
        upButton={mockUpButton}
        downButton={mockDownButton}
      />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("renders correctly when a user has not registered any gear to a favoriteList yet", () => {
    const renderResult = render(
      <GearCard
        gear={testGear}
        addButton={mockAddButton}
        removeButton={mockRemoveButton}
        upButton={mockUpButton}
        downButton={mockDownButton}
      />
    );

    expect(renderResult.container.firstElementChild).toHaveAttribute(
      "class",
      expect.stringContaining("root")
    );

    const imgElement = renderResult.getByRole("img");
    expect(imgElement).toHaveAttribute(
      "src",
      expect.stringMatching("mediumImageUrl")
    );
    expect(imgElement).toHaveAttribute(
      "alt",
      expect.stringMatching("productName")
    );
    expect(
      renderResult.container.getElementsByClassName(
        expect.stringContaining("avatar")
      )
    ).toBeTruthy();

    expect(renderResult.getByText("productName")).toBeInTheDocument();
    expect(renderResult.getByText(`${testGear.makerName}`)).toBeInTheDocument();

    const buttonElements = renderResult.getAllByRole("button");
    expect(buttonElements).toHaveLength(5);
    expect(buttonElements[4]).toHaveAttribute(
      "href",
      expect.stringMatching(testGear.affiliateUrl)
    );
  });

  it("renders correctly when user's favoriteList contains same gear", () => {
    const renderResult = render(
      <GearCard
        gear={testFavoriteList.gears[0]}
        favoriteList={testFavoriteList.gears}
        addButton={mockAddButton}
        removeButton={mockRemoveButton}
        upButton={mockUpButton}
        downButton={mockDownButton}
      />
    );

    const buttonElements = renderResult.getAllByRole("button");
    expect(buttonElements[2]).toHaveTextContent("My Favorite Gearに登録済み");
    expect(buttonElements[2]).toBeDisabled();
  });

  it("triggers user events", async () => {
    const renderResult = render(
      <GearCard
        gear={testGear}
        addButton={mockAddButton}
        removeButton={mockRemoveButton}
        upButton={mockUpButton}
        downButton={mockDownButton}
      />
    );

    const buttonElements = renderResult.getAllByRole("button");
    const upButton = buttonElements[0];
    const downButton = buttonElements[1];
    const addButton = buttonElements[2];
    const removeButton = buttonElements[3];

    userEvent.click(upButton);
    expect(mockUpButton).toBeCalled();

    userEvent.click(downButton);
    expect(mockDownButton).toBeCalled();

    userEvent.click(addButton);
    expect(mockAddButton).toBeCalled();

    userEvent.click(removeButton);
    expect(mockRemoveButton).toBeCalled();
  });
});
