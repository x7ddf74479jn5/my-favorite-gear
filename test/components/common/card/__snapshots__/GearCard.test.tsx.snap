// Vitest Snapshot v1

exports[`GearCard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3kfr4u-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-85e2a5-MuiAvatar-root"
        >
          <img
            alt="productName"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="mediumImageUrl"
          />
        </div>
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          productName
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-ptbrmz-MuiTypography-root"
        >
          makerName/brandName/genreName
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="css-gg4vpm"
      >
        <div
          class="css-k008qs"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             : 1
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-hlnzui-MuiTypography-root"
        >
          平均価格: averagePrice円
        </p>
      </div>
      <div
        class="css-1yj626v"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth css-x6yic9-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
          href="https://www.affiliate.com"
          rel="noopener"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </span>
          楽天で見る
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth css-1yf2m96-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
          href="https://www.amazon.com/"
          rel="noopener"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </span>
          Amazonで見る
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowUpwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDownwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-1jyq7yo-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ThumbUpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
            />
          </svg>
        </span>
        My Favorite Gearに登録
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-disableElevation css-19bi5f5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        My Favorite Gearから削除
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GearCard > matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3kfr4u-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-85e2a5-MuiAvatar-root"
        >
          <img
            alt="productName 1"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="mediumImageUrl 1"
          />
        </div>
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          productName 1
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-ptbrmz-MuiTypography-root"
        >
          makerName 1/brandName 1/genreName 1
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="css-gg4vpm"
      >
        <div
          class="css-k008qs"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
             : 1
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-hlnzui-MuiTypography-root"
        >
          平均価格: averagePrice 1円
        </p>
      </div>
      <div
        class="css-1yj626v"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth css-x6yic9-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
          href="https://www.affiliate.com/1"
          rel="noopener"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </span>
          楽天で見る
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth css-1yf2m96-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
          href="https://www.amazon.com/1"
          rel="noopener"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </span>
          Amazonで見る
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowUpwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDownwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-1jyq7yo-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ThumbUpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
            />
          </svg>
        </span>
        My Favorite Gearに登録済み
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-disableElevation css-19bi5f5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        My Favorite Gearから削除
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
